build-farm/make-adopt-build-farm.sh

```diff
@@ -18,16 +18,40 @@ set -e
 
 PLATFORM_SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-## Sanity check as this script requires various environment variables so fail
-## fast if one or more are not present. MAYBE autodetect defaults in future?
+## autodetect defaults to improve usability when running this for debugging/testing
+
+if [ -z "$ARCHITECTURE"  ]; then
+   ARCHITECTURE=`uname -p`
+   if [       "$OSTYPE" = "cygwin"  ]; then ARCHITECTURE=`uname -m`; fi
```

Comment:
```suggestion
   if [ "$OSTYPE" = "cygwin"  ]; then ARCHITECTURE=`uname -m`; fi
```

Reply:
Hmmm I disagree with this - better to have the `[ ]; then ARCHITECTURE=` lined up for readability

Reply:
Why not increase the indentation of the others to match for it in that case? I'll approve this regardless as it's just a suggestion

Reply:
Personal preference I guess - I definitely prefer:
```
   if [       "$OSTYPE" = "cygwin"  ]; then ARCHITECTURE=`uname -m`; fi
   if [ "$ARCHITECTURE" = "x86_64"  ]; then ARCHITECTURE=x64;        fi
   if [ "$ARCHITECTURE" = "powerpc" ]; then ARCHITECTURE=ppc64;      fi
```
over 
```
if [ "$OSTYPE" = "cygwin"        ]; then ARCHITECTURE=`uname -m`; fi
if [ "$ARCHITECTURE" = "x86_64"  ]; then ARCHITECTURE=x64;        fi
if [ "$ARCHITECTURE" = "powerpc" ]; then ARCHITECTURE=ppc64;      fi
```
because the `=` are lined up, so I think I'll leave it, but noted for the future ;-)

Reply:
(I suppose the third option would be:
```
if [ "$OSTYPE"       = "cygwin"  ]; then ARCHITECTURE=`uname -m`; fi
if [ "$ARCHITECTURE" = "x86_64"  ]; then ARCHITECTURE=x64;        fi
if [ "$ARCHITECTURE" = "powerpc" ]; then ARCHITECTURE=ppc64;      fi
```
Hmmmm ....

Reply:
OK I'll compromise on that last one :-) Updated